apps/client/src/routes/edge-login/edge-login-page.tsx

```diff
@@ -0,0 +1,139 @@
+import { useState } from 'react';
+import Form from '@cloudscape-design/components/form';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Button from '@cloudscape-design/components/button';
+import Container from '@cloudscape-design/components/container';
+import Header from '@cloudscape-design/components/header';
+import FormField from '@cloudscape-design/components/form-field';
+import Input from '@cloudscape-design/components/input';
+import Select from '@cloudscape-design/components/select';
+import Box from '@cloudscape-design/components/box';
+import ColumnLayout from '@cloudscape-design/components/column-layout';
+import { colorBackgroundHomeHeader } from '@cloudscape-design/design-tokens';
+import { authService } from '../../auth/auth-service';
+import { PropsWithChildren } from 'react';
+import { useEdgeLoginQuery } from './hooks/use-edge-login-query';
+
+// Define an empty type to use PropsWithChildren<EdgeLoginProps>
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface EdgeLoginProps {}
+
+export function EdgeLoginPage({ children }: PropsWithChildren<EdgeLoginProps>) {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [ipAddress, setIpAddress] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [authMechanism, setAuthMechanism] = useState({
+    label: 'Linux',
+    value: 'linux',
+  });
+  const [error, setError] = useState('');
+  const { refetch } = useEdgeLoginQuery({
+    edgeEndpoint: ipAddress,
+    username,
+    password,
+    authMechanism: authMechanism.value,
+  });
+
+  const getCredentials = async () => {
+    const { data } = await refetch();
+    if (data) {
+      authService.setAwsCredentials(data);
+      setIsLoggedIn(true);
+    } else {
+      setError('Error logging in');
+    }
+  };
+  if (isLoggedIn) {
+    return children;
+  } else {
+    return (
+      <ColumnLayout columns={3}>
+        <div></div>
```

Comment:
I wanted to display the login page in the middle of the screen, so I created a 3 column layout with two empty divs and the login page in the middle. I wanted to avoid using CSS but maybe should just use CSS for this case. I found this way of doing it easy to understand but it might not be obvious what ColumnLayour is doing here. What do you think?